storyblok/storyblok-js-client
# fix: fixes cases re throttledPromised: abort misbehaviour and not awaited queue

We've been reported in #682, and thanks to the discovery of @dojscart in https://github.com/withastro/adapters/issues/43#issuecomment-1782206087 (really really thanks 🙏 ) we've managed to identify that the internal queue of the throttle promise wasn't being awaited properly, causing some edge cases to fail (like Cloudflare, and likely the SSG generation).

On the way, I also managed to find another case -> after aborting the queue, promises kept being added and got stuck in the queue unresolved forever.

**Solution**
Basically, now the promises in the queue are awaited properly after crossing the limit, and no new promises are getting added after aborting the queue. 

Fixes #682 

### How to test the issue

- Run the new test `npx vitest run src/throttlePromise.test.ts` against the `src/throttlePromise.ts` living in the `main` branch -> **both test cases should fail**
- Now, run again the same test against the updated `src/throttlePromise.ts` from this branch -> **both test cases should pass correctly**

Aside from that, I can recommend to try to break the expected functionality of the throttlePromise and its internal queue. I tried via a local script to heavily load the queue with all kind of scenarios - thousands of calls, randomly make them fail, etc - and so far it worked out, but extra eyes and hands are always great.
